data-protocols/ids/ids-spi/src/main/java/org/eclipse/dataspaceconnector/ids/spi/types/IdsId.java

```diff
@@ -66,6 +72,40 @@ public IdsId build() {
         }
     }
 
+    public @Nullable URI toUri() {
+        try {
+            return URI.create(String.join(DELIMITER, SCHEME, this.getType().getValue(), this.getValue()));
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    public static @Nullable IdsId from(URI uri) {
+        try {
+            return from(uri.getScheme() + DELIMITER + uri.getSchemeSpecificPart());
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    public static IdsId from(String urn) {
+        if (urn == null) {
+            throw new IllegalArgumentException("String must not be null");
```

Comment:
Right, but direct method calls provide exception handling. I also came across this but copied it as-is. Result sounds like a good solution. I will try and see if this can be replaced easily.